addons/point_of_sale/models/pos_order.py

```diff
@@ -446,6 +448,12 @@ def create(self, vals_list):
         for vals in vals_list:
             session = self.env['pos.session'].browse(vals['session_id'])
             vals = self._complete_values_from_session(session, vals)
+            related_country = False
+            if vals.get('mobile'):
+                related_country = self.env['res.partner'].browse(vals['partner_id']).country_id \
+                    if vals.get('partner_id') else self.env.company.country_id
+                vals['mobile'] = self._phone_format(number=vals.get('mobile'),
+                        country=related_country)
```

Comment:
maybe instead of that you can create a compute methods that depends on the partner.

Reply:
btw you don't need it if you set the vals from the frontend